src/components/log/LogContainer.tsx

```diff
@@ -41,25 +43,25 @@ export class LogContainer extends React.Component<Props, State> {
   }
 
   public componentDidUpdate() {
-    if (this.props.task.logs !== undefined && this.state.logText === "") {
+    if ((this.props.task.logs !== undefined && this.state.logText === "") || (this.props.task.task_id !== this.state.taskId)) {
       this.props.client.getLogs((err, resp, body) => {
         this.setState({
+          taskId: this.props.task.task_id,
           logText: body,
+          logType: LogType.task,
           htmlLink: this.props.task.logs.task_log,
           rawLink: this.props.task.logs.task_log + "&text=true"
         });
-      }, this.props.task.task_id, LogType.task, this.props.task.execution)
+      }, this.props.task.task_id, LogType.task, this.props.task.execution);
     }
   }
 
   public render() {
 
-    // TODO: use Evergreen API to determine the severity of each log line (see example below)
-    // (https://evergreen.mongodb.com/json/task_log/spruce_ubuntu1604_lint_37ec5816d14e05bd02535788d114da8b97ed0231_19_07_11_18_09_18/0?type=T)
     const Logs = () => (
       <InfiniteScroll loadMore={this.dummyLoadMore} className="log-scrollable">
         {this.state.logText.split("\n").map(textLine => (
-          <Typography className={this.lineContainsError(textLine.toLowerCase()) ? "log-text-error" : "log-text-normal"}>
+          <Typography className={this.lineContainsError(textLine.toLowerCase()) ? "log-text-error" : "log-text-normal"} key={new Date().getTime().toString()}>
```

Comment:
I was getting lots of console warnings about the log lines not having unique IDs. I decided to use datetime as the basis for a unique key at the suggestion of a StackOverflow post (using an incrementing number as you iterate through the list is a bad idea apparently). Let me know if you have a better/less strange solution.

Reply:
I'd prefer to not have any key since I see little need to do so. Do you know why a typography component needs a key?

Reply:
It's not actually the typography component that needs it. This is a general requirement of using the map() function to generate elements in React. Things still work without unique keys, it just fills the console with messages warning that elements should have a unique key. You can read more about that [here](https://reactjs.org/docs/lists-and-keys.html).